### PR TITLE
Add missing parallel algorithms to algorithm.hpp

### PR DESCRIPTION
**Pull request by Syntaf**
I'm not sure if this was intentional, but there are a number of algorithms missing from the `algorithm.hpp` include. This affects `#include <hpx/include/parallel_algorithm.hpp>` , as this include calls `hpx/parallel/aglorithm.hpp`

### CODE REVIEW

File: docs/manual/build_system/cmake_variables.qbk [hunk 1]
```diff
@@ -34,6 +34,7 @@ The options are split into these categories:
 * [link build_system.cmake_variables.HPX_WITH_COMPRESSION_BZIP2 HPX_WITH_COMPRESSION_BZIP2]
 * [link build_system.cmake_variables.HPX_WITH_COMPRESSION_SNAPPY HPX_WITH_COMPRESSION_SNAPPY]
 * [link build_system.cmake_variables.HPX_WITH_COMPRESSION_ZLIB HPX_WITH_COMPRESSION_ZLIB]
+* [link build_system.cmake_variables.HPX_WITH_FORTRAN HPX_WITH_FORTRAN]
```
Comment (Syntaf) [inline docs/manual/build_system/cmake_variables.qbk]: @hkaiser @sithhell How can I prevent this from happening when I commit changes? It seems building the HPX docs will produce stuff like this

File: hpx/parallel/algorithm.hpp [hunk 2]
```diff
@@ -38,5 +39,7 @@
 #include <hpx/parallel/algorithms/set_union.hpp>
 #include <hpx/parallel/algorithms/sort.hpp>
 #include <hpx/parallel/algorithms/swap_ranges.hpp>
+#include <hpx/parallel/algorithms/uninitialized_copy.hpp>
+#include <hpx/parallel/algorithms/uninitialized_fill.hpp>
```
Comment (hkaiser) [inline hpx/parallel/algorithm.hpp]: We have those in `hpx/parallel/memory.hpp`, just like the original `uninitialized_xxx` algorithms. The TS is not very specific on what algorithms should be defined by which header, thus we decided to stay as close to the sequential algorithms.

File: docs/manual/build_system/cmake_variables.qbk (hunk 1 above)
Comment (hkaiser) [inline docs/manual/build_system/cmake_variables.qbk]: Just don't commit those files ;-)

**Comment by hkaiser** [comment]
@Syntaf I think the 
```
 #include <hpx/parallel/algorithms/includes.hpp>
```
was missing. Let's add it.

**Pull request closed without merging**